packages/block-library/src/buttons/edit.native.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	__experimentalAlignmentHookSettingsProvider as AlignmentHookSettingsProvider,
+} from '@wordpress/block-editor';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, useResizeObserver } from '@wordpress/compose';
+import { createBlock } from '@wordpress/blocks';
+import { useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { name as buttonBlockName } from '../button/';
+import styles from './editor.scss';
+
+const ALLOWED_BLOCKS = [ buttonBlockName ];
+const BUTTONS_TEMPLATE = [ [ 'core/button' ] ];
+
+function ButtonsEdit( {
+	isSelected,
+	attributes,
+	onDelete,
+	onAddNextButton,
+	shouldDelete,
+	isInnerButtonSelected,
+} ) {
+	const { align } = attributes;
+	const [ resizeObserver, sizes ] = useResizeObserver();
+	const [ maxWidth, setMaxWidth ] = useState( 0 );
+	const shouldRenderFooterAppender = isSelected || isInnerButtonSelected;
+	const { marginLeft: spacing } = styles.spacing;
+
+	useEffect( () => {
+		const margins = 2 * styles.parent.marginRight;
+		const { width } = sizes || {};
+		if ( width ) {
+			setMaxWidth( width - margins );
+		}
+	}, [ sizes ] );
+
+	function renderAppender() {
+		if ( shouldRenderFooterAppender ) {
```

Comment:
Do we need that check if we have it already here - https://github.com/WordPress/gutenberg/pull/20191/files#diff-260b0f5b413f5ac735a6935b2f8acc34R271